web/src/filter.ts

```diff
@@ -901,7 +912,7 @@ export class Filter {
             const dm_operand = dm_participants.join(",");
 
             const dm_conversation_terms = [{operator: "dm", operand: dm_operand, negated: false}];
-            return [...dm_conversation_terms, ...filtered_terms];
+            return z.array(narrow_term_schema).parse([...dm_conversation_terms, ...filtered_terms]);
```

Comment:
Unnecessary runtime parsing. TypeScript understands the type already, with `as const`:

```ts
const dm_conversation_terms = [
    {operator: "dm", operand: dm_operand, negated: false},
] as const;
return [...dm_conversation_terms, ...filtered_terms];
````

Or with a type annotation:

```ts
const dm_conversation_terms: NarrowTerm[] = [
    {operator: "dm", operand: dm_operand, negated: false},
];
return [...dm_conversation_terms, ...filtered_terms];
```

Or inline, to take advantage of the function’s return type annotation:

```ts
return [{operator: "dm", operand: dm_operand, negated: false}, ...filtered_terms];
```